### ISSUE TITLE
Rename de game.whenCollideDo

### ISSUE THREAD

**Issue by asanzo** [uqbar-project/wollok-language#36]
Me parece importante tener una buena separación en el **nombre** y también en el "javadoc" entre `onCollideDo` y el actual `whenCollideDo`. Hoy son casi idénticas las documentaciones y casi idénticos los nombres cuando en realidad el primero lo hace muchas veces y el segundo sólo una vez.

Propongo renombrar el `game.whenCollideDo` a `game.whileCollideRepeat`, y agregar a las respectivas documentaciones la comparación y alguna cosita más que falta:

```
whenCollideDo
Adds a block that will be executed once when the given object collides with other.
Two objects collide when are in the same position.

The block should expect the other object as parameter.

Example:
game.onCollideDo(pepita, { comida => pepita.comer(comida) })

If there are more objects colliding, the block will be executed once for each collider. 

Compare with whileCollideRepeat.
```

```
whileCollideRepeat
Adds a block that will be executed repeatedly while the given object is colliding with other.
Two objects collide when are in the same position.

The block should expect the other object as parameter. Example:
game.whileCollideRepeat(pepita, { comida => pepita.comer(comida) })

If there are more objects colliding, the block will be executed repeatedly for each collider. 

Compare with onCollideDo.
```

**Comment by fdodino** [comment]
@PalumboN por ahora veo que sigue `whenCollideDo`, avanzamos con el ticket?

**Comment by PalumboN** [comment]
Sí, avancemos. Esto va a romper los juegos actuales :( No tenemos ningún mecanismo de deprecación todavía no? Da para pensar algo de eso?

**Comment by fdodino** [comment]
No tenemos mecanismo de deprecación, si querés nos juntamos algún día y pensamos en algo...
Podemos inaugurarlo con este caso...

**Comment by asanzo** [comment]
Me fumé un caño e hice un documento para hablar de la deprecación, como si tuviera un ejército de 100 copias de Dodino para mí:
https://docs.google.com/document/d/1vWqL5N7LgE6gSD2n4kAKlvpyb7LqLwNA_rjVvTa7xi4/edit#
(Rastas necesitamos uno solo: por alguna razón revisar un PR lo puede hacer en tiempo negativo, me sorprende)

En todo caso, se puede mandar todo al carajo, sólo quería crear el doc y no dejarlo en blanco XD.
Puedo mandar msj a wollok-dev si les va.

**Comment by fdodino** [comment]
es el camino @asanzo , tener un documento detallado nos permite saber cuáles son las ideas y ver qué tiempo tenemos para implementarlas. Muy groso!

**Comment by lspigariol** [comment]
no es un poco fuerte deprecarlo por un cambio de nombre?
no lo tengo claro, pero me suena que deprecar es un recurso extraordinario
por cuestiones de fuerza mayor, por un motivo grave, que cause
algún problema, que traiga consecuencias no deseadas.
se me ocurre alguna recomendacion clara de no uso, indicación de uso
alternativo, y en el mejor de los casos una adecuación progresiva de los
ejemplos existentes.

**Comment by PalumboN** [comment]
Yo tiré la de deprecación porque pensé que la idea era cambiar el nombre del método. Sino lo que se puede hacer es dejar los dos (como hacemos con `size` y `length` si mal no recuerdo). 
Pero dado que en este caso es por un tema de expresividad, pensando en los usuarios, me parecía mejor no dejar los 2 métodos. En realidad esto sería lo ideal, ya que tener varios métodos para lo mismo puede resultar confuso y hace que diverjan los ejemplos y otras cosas.

-----

Por otro lado, llevaría esta nomenclatura a otros métodos, como el `onTick`, que debería ser `whileTickRepeat` o algo por el estilo. El `onPressDo` estaría bien.

**Comment by fdodino** [comment]
Lo de la deprecación va en paralelo, en algún momento pensamos en tener un migrador, estamos todavía lejos de eso. 

A mí personalmente me gustaría depurar los nombres, se que es molesto para los que hicieron los juegos, pero también me parece que una forma de avanzar es tener una interfaz que sea entendible y que vaya mejorando.

**Comment by PalumboN** [comment]
En la reunión de Wollok definimos que el `whenCollide` se borra :)
